src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java

```diff
@@ -153,13 +154,26 @@ public void setEnvVars(List<ContainerEnvVar> envVars) {
         this.envVars.addAll(envVars);
     }
 
+
     public ContainerLivenessProbe getLivenessProbe() { return livenessProbe; }
 
     @DataBoundSetter
     public void setLivenessProbe(ContainerLivenessProbe livenessProbe) {
         this.livenessProbe = livenessProbe;
     }
 
+    public List<PortMapping> getPorts() {
+        return ports != null ? ports : Collections.emptyList();
+    }
+
+    @DataBoundSetter
+    public void setPorts(List<PortMapping> ports) {
+        if (ports != null) {
+            this.ports.clear();
+            this.ports.addAll(ports);
```

Comment:
the way jenkins (de)serializes is a bit confusing. In upgrades this.ports will be null because the field is not in the xml config.
What you want is to guard against that, so just remove `final` (I don't think there's any reason to have it) and do this.ports=ports